org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/XMLLanguageServer.java

```diff
@@ -230,6 +230,7 @@ private synchronized void updateSettings(Object initOptions, boolean initLogs) {
 		if (capabilityManager.getClientCapabilities().getExtendedCapabilities().shouldLanguageServerExitOnShutdown()) {
 			delayer.schedule(() -> exit(0) , 1, TimeUnit.SECONDS);
 		}
+		getTelemetryManager().shutdown();

```

Comment:
This should happen in a separate thread that doesn't affect the shutdown process.